src/tests/test_parse.py

```diff
@@ -505,3 +506,17 @@ def test_escape_string_slicing_notation(query, expected_escaped, expected_found)
     escaped, found = escape_string_slicing_notation(query)
     assert escaped == expected_escaped
     assert found == expected_found
+
+
+@pytest.mark.parametrize(
```

Comment:
Please add more tests for non-CTE queries also which have different combinations of parenthesis

Reply:
@neelasha23 This test just checks for valid parentheses, i.e. if each opening parentheses has a correct corresponding closing parentheses. It works regardless of the query being CTE or not.